lib/logger.js

```diff
@@ -72,6 +74,26 @@ function now () {
   return (ts[0] * 1e3) + (ts[1] / 1e6)
 }
 
+function createLogger (options) {
+  if (isValidLogger(options.logger)) {
+    const logger = createPinoLogger({
+      logger: options.logger,
+      serializers: Object.assign({}, serializers, options.logger.serializers)
+    })
+    return [logger, true]
+  } else if (!options.logger) {
+    const logger = Object.create(abstractLogging)
+    logger.child = () => logger
+    return [logger, false]
```

Comment:
Using an object instead of an array would make this code easier to understand. With the array, it's not clear what `false` means here.

Reply:
Thank you for the feedback, but for now I'll leave it as is. By reading the code is easy to understand what that boolean is, and if this becomes a problem in the future I'll add some more doc.
Furthermore this syntax is becoming commonly used in JavaScript, so let's start embracing it :)

Reply:
I agree with @nwoltman. Just reading this function, I can't begin to tell you how this result will be used. Going over to `fastify.js` I can see that it is meant to be `{logger, hasLogger}`. I think this function should `return {logger, hasLogger: false}`.

See https://docs.google.com/document/d/1hWb-lQW4NSG9yRpyyiAA_9Ktytd5lypLnVLhPX9vamE/edit for information on array destructuring. I'm not opposed to array destructuring because of the doc, but it's something to keep in mind. I am simply opposed here due to legibility.